packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -239,14 +239,21 @@ class IOSDevices extends PollingDeviceDiscovery {
   @override
   Future<void> stopPolling() async {
     await _observedDeviceEventsSubscription?.cancel();
+    _observedDeviceEventsSubscription = null;
+    xcdevice.cancelWirelessDiscovery();
```

Comment:
You shouldn't use `stopPolling` because this will also kill IDE device discovery. I recommend making a new in src/device.dart that does something like

```dart
void stopExtendedWirelessDeviceDiscoverers() {
    final IOSDevices? iosDiscoverer = _platformDiscoverers.whereType<IOSDevices>().firstOrNull;
    await iosDiscoverer?.cancelWirelessDiscovery();
  }
```